react-table/src/ConfigurableTable/ConfigurableTable.tsx

```diff
@@ -286,11 +292,17 @@ function ColumnSelection(props: IColSelectionProps) {
                                 Record={c}
                                 Setter={() => props.onChange(c.Key)}
                                 key={c.Key}
-                                Disabled={c.Key == props.sortKey || (props.disableAdd && !c.Enabled)}
+                                Disabled={
+                                    c.Key === props.sortKey ||
+                                    (props.disableAdd && !c.Enabled) ||
+                                    (isOnlyOneEnabled && c.Enabled)
+                                }
                                 Help={
-                                    c.Key == props.sortKey
+                                    c.Key === props.sortKey
                                         ? 'The Table is currently sorted by this column so it cannot be hidden.'
-                                        : undefined
+                                    : isOnlyOneEnabled && c.Enabled
+                                        ? 'Table must have one column visible at all times.'
+                                    : undefined
```

Comment:
This seems unneccesarrily complex to read. Maybe add a `helperMessage: String|undefined` using `React.Memo`

Reply:
Moved to it's own function